enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,439 @@
+and exclusive to a management component ([same as today](https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go#L592)).
+While the `shared` cpus are dedicated for workloads which are asking for shared cpus.
+
+The following shows the cpu layout and what processes can be running at what cpus:
+![cpu layout](cpu_layout.jpg)
+Please check the [Risks and Mitigations](mixed-cpu-node-plugin.md#risks-and-mitigations) for additional information.
+
+There is a plan for a feature called [shared-partition](https://github.com/openshift/enhancements/pull/1421) 
+that implements additional pool.
+When shared-partition lands, Kubelet's cpu pool layout should look roughly like:
+1. `reservedSystemCpus`
+2. `guaranteedCpus` <- new pool
+3. `sharedCpus` = All CPUs - `reservedSystemCpus` - `guaranteedCpus`
+
+When shared-partition feature enabled, 
+mixed-cpu-node-plugin allocates the shared cpus from the `sharedCpus` pool and not
+from the `reservedSystemCpus` anymore.
+
+The following table shows the shared-cpus origin in each flow:
+
+| mixed-cpu | shared-partition | shared cpus origin                 |
+|-----------|------------------|------------------------------------|
+| Enable    | Disable          | Kubelet's `reservedSystemCPUs`     | 
+| Disable   | Enable           | All cpus - `reserved` - `isolated` | 
+| Enable    | Enable           | All cpus - `reserved` - `isolated` | 
+
+#### Workload configuration
+A workload that wants to access the shared cpu should request for `openshift.io/enabled-shared-cpus` under its spec.
+The `openshift.io/enabled-shared-cpus` should be treated as a boolean value.
+In other words, the resource request only uses as a hint that shared-cpus required for the container and
+does not indicate any actual value.
+
+For example:
+```yaml
+requests:
+   openshift.io/enabled-shared-cpus: 1
+```
+Fine.
+
+```yaml
+requests:
+   openshift.io/enabled-shared-cpus: 2
+```
+Wrong. An error will be returned to the user explaining how to fix the pod spec.
+
+The reason for specifying a resource request, and not an annotation,
+is to allow control on the number of workloads that can request for shared cpus.
+By populating shared cpu resources as the number of maximum workloads.
+If the number of shared cpu requests got exhausted, any new workload that requests for shared cpus request will move
+to pending state by the scheduler.
+
+Once workload request completed successfully it gets access to shared cpus, and two new environment variables
+will be present under the container's environment variables:
+`OPENSHIFT_SHARED_CPUS=<CPU-IDs>` = specifies the core ids of the shared cpus
+`OPENSHIFT_ISOLATED_CPUS=<CPU-IDs>` = specifies the core ids of the isolated cpus
+Those environment variables help the application's user to pin its processes/threads to the desired CPU set. 
```

Comment:
For v2, we should also expose information about the child cgroup to join.

Reply:
It doesn't have to be configurable, we can use the same child cgroup name for every workloads.
We would need to provide separate procedure (sample app, documentation, technical enablement, etc.) about how to join the cgroup. WDYT?

Reply:
Yeah, we need documentation for sure. We can decide if we want to add an env var as convenience as part of the implementation.